ClassHierarchy_hw/Classes/Ingredient.cs

```diff
@@ -0,0 +1,55 @@
+﻿// <copyright file="Ingredient.cs" company="PlaceholderCompany">
+// Copyright (c) PlaceholderCompany. All rights reserved.
+// </copyright>
+
+namespace ClassHierarchy_hw.Classes
+{
+    internal class Ingredient : IngredientType
+    {
+        private string name;
+        private double calories;
+        private double quantity;
+        private string measurement;
+        private string type;
+
+        public Ingredient(string name, double calories, double quantity, string measurement, string type)
+            : base(name)
+        {
+            this.Name = name;
+            this.Quantity = quantity;
+            this.Calories = calories;
+            this.Measurement = measurement;
+            this.Type = type;
+        }
+
+        public new string Name
+        {
+            get => this.name;
```

Comment:
Since there is no logic inside getters and setters - you can use automatic properties